website/client/src/components/header/notifications/groupTaskApproved.vue

```diff
@@ -15,17 +15,129 @@
 </template>
 
 <script>
+import axios from 'axios';
+import Vue from 'vue';
 import BaseNotification from './base';
+import scoreTask from '@/../../common/script/ops/scoreTask';
+import { mapState } from '@/libs/store';
 
 export default {
   components: {
     BaseNotification,
   },
   props: ['notification', 'canRemove'],
+  computed: {
+    ...mapState({
+      user: 'user.data',
+    }),
+  },
   methods: {
-    action () {
-      const { groupId } = this.notification.data;
-      this.$router.push({ name: 'groupPlanDetailTaskInformation', params: { groupId } });
+    async action () {
```

Comment:
can we share some of this code with https://github.com/HabitRPG/habitica/blob/develop/website/client/src/components/tasks/task.vue#L1024 ?

Reply:
@negue before the PR gets merged can we address this problem?

Reply:
@SabreCat will extract it  - so that we don't have the duplicate code

Reply:
How's this look? https://github.com/HabitRPG/habitica/pull/12142/commits/584b0b1f9c77251bca70457906a94218f5d81353 (the couple of CSS bits are unrelated)

Reply:
looks good!